modules/rosa-cluster-hcp/main.tf

```diff
@@ -0,0 +1,63 @@
+data "aws_caller_identity" "current" {}
+
+data "aws_region" "current" {}
+
+locals {
+  sts_roles = {
+    role_arn         = var.installer_role_arn,
+    support_role_arn = var.support_role_arn,
+    instance_iam_roles = {
+      worker_role_arn = var.worker_role_arn
+    },
+    operator_role_prefix = var.operator_role_prefix,
+    oidc_config_id       = var.oidc_config_id
+  }
+}
+
+data "aws_subnet" "provided_subnet" {
+  count = length(var.aws_subnet_ids)
+
+  id = var.aws_subnet_ids[count.index]
+}
+
+resource "rhcs_cluster_rosa_hcp" "rosa_sts_cluster" {
+  name           = var.cluster_name
+  cloud_region   = data.aws_region.current.name
+  aws_account_id = data.aws_caller_identity.current.account_id
+  aws_billing_account_id = var.aws_billing_account_id == "" ? data.aws_caller_identity.current.account_id : var.aws_billing_account_id
+  replicas       = var.replicas
+  version        = var.openshift_version
+  properties = {
+    rosa_creator_arn = data.aws_caller_identity.current.arn
+  }
+  sts            = local.sts_roles
+  aws_subnet_ids = var.aws_subnet_ids
+  # availability_zones = slice(data.aws_availability_zones.available.names, 0, var.multi_az ? 3 : 1)
+  availability_zones = length(var.availability_zones) > 0 ? (
+    var.availability_zones
+    ) : (
+    length(var.aws_subnet_ids) > 0 ? (
+      distinct(data.aws_subnet.provided_subnet[*].availability_zone)
+      ) : (
+      slice(data.aws_availability_zones.available.names, 0, 3)
+    )
+  )
+  compute_machine_type = var.compute_machine_type
+  #autoscaling_enabled  = var.autoscaling_enabled
+  #min_replicas         = var.min_replicas
+  #max_replicas         = var.max_replicas
+  machine_cidr         = var.machine_cidr
+
+  wait_for_create_complete = true
```

Comment:
Will look into it, considering the standard compute pool has a status exposed I suppose it should be doable

Reply:
It will be something like this:
```
func pollClusterCompute(clusterId string, ctx context.Context, timeout int64, clusterCollection *cmv1.ClustersClient) (*cmv1.Cluster, error) {
	client := clusterCollection.Cluster(clusterId)
	var object *cmv1.Cluster
	pollCtx, cancel := context.WithTimeout(ctx, time.Duration(timeout)*time.Minute)
	defer cancel()
	_, err := client.Poll().
		Interval(pollingIntervalInMinutes * time.Minute).
		Predicate(func(getClusterResponse *cmv1.ClusterGetResponse) bool {
			object = getClusterResponse.Body()
			tflog.Debug(ctx, "polled cluster compute", map[string]interface{}{
				"currentCompute": object.Status().CurrentCompute(),
			})
			switch object.Status().CurrentCompute() {
			case object.Nodes().Compute():
				return true
			}
			return false
		}).
		StartContext(pollCtx)
	if err != nil {
		tflog.Error(ctx, "Failed polling cluster compute")
		return nil, err
	}

	return object, nil
}
```

Reply:
Looking internally it seems to pickup from the node pools anyways, so yeah something like that would suffice indeed. I wonder if we should start to move away from picking up it from the cluster though and look actively into the pool though

Reply:
Added to https://github.com/terraform-redhat/terraform-provider-rhcs/pull/553